### PR TITLE
=htp #18615 rm comment on workaround in Scala 2.10

### LINKED ISSUE #18615: Remove warning on FormFieldDirectives once on Scala 2.11+ only
Once we merge with 2.4, i.e. streams only supports 2.11+ in OSS we should drop this message from FormFieldDirectives:

```
   * Due to a bug in Scala 2.10, invocations of this method sometimes fail to compile with an
   * "too many arguments for method formField" or "type mismatch" error.
```

### PR DESCRIPTION
**Pull request by ktoso**
Resolves #18615

### CODE REVIEW

File: akka-http/src/main/scala/akka/http/scaladsl/server/directives/FormFieldDirectives.scala [hunk 1]
```diff
@@ -20,7 +20,7 @@ trait FormFieldDirectives extends ToNameReceptacleEnhancements {
   import FormFieldDirectives._
 
   /**
-   * Extracts HTTP form fields from the request as a ``Map[String, String]``.
+   * Extracts HTTP form fields from the request g in Scala 2.10, invocations of this method sometimes fail to as a ``Map[String, String]``.
```
Comment (patriknw) [inline akka-http/src/main/scala/akka/http/scaladsl/server/directives/FormFieldDirectives.scala]: I don't understand this sentence. There is a `g` and what does Scala 2.10 has to do with us?

Comment (ktoso) [inline akka-http/src/main/scala/akka/http/scaladsl/server/directives/FormFieldDirectives.scala]: Whoa, how did this get here... :-\
PR is only about the removals below (will remove this one)

**Comment by ktoso** [comment]
PR fixed – was LGTM otherwise? :)

**Comment by patriknw** [comment]
LGTM

**Pull request merged**